library/src/main/java/com/novoda/downloadmanager/DownloadManagerBuilder.java

```diff
@@ -0,0 +1,33 @@
+package com.novoda.downloadmanager;
+
+import android.content.ContentResolver;
+
+import com.novoda.downloadmanager.lib.DownloadManager;
+
+public class DownloadManagerBuilder {
+
+    private ContentResolver contentResolver;
+    private boolean verboseLogging;
+
+    public static DownloadManagerBuilder create() {
+        return new DownloadManagerBuilder();
+    }
+
+    public DownloadManagerBuilder withVerboseLogging() {
+        this.verboseLogging = true;
+        return this;
+    }
+
+    public DownloadManagerBuilder with(ContentResolver contentResolver) {
```

Comment:
I still personally think Joe Bloggs would be more used to seeing constructors than factory methods... just we here aren't because we hide them with `newInstance`. But don't get me started on `newInstance` either :tongue: